### PR TITLE
DAOS-10132 doc: Describe disable_vfio usage in admin guide

### PR DESCRIPTION
**Pull request by tanabarr**
Include information on setting option when using daos_server storage
prepare as well as setting the parameter in the server config file.

Doc-only: true

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: docs/admin/deployment.md [hunk 1]
```diff
@@ -485,14 +485,19 @@ storage selection.
...
+`dmg storage scan` can be run to query remote running `daos_server` processes over the management
+network.
+
+`daos_server storage scan` can be used to query storage on local hosts directly.
+
+NVMe SSDs need to be made accessible first by running `daos_server storage prepare --nvme-only`.
+The default way for DAOS to access NVMe storage is through SPDK via the VFIO user-space driver.
+To use an alternative driver with SPDK, set `--disable-vfio` in the storage prepare command to
+fallback to using UIO user-space driver with SPDK instead. The option is only effective when
+IOMMU and VFIO are enabled on the host.
```
Comment (kjacque) [inline docs/admin/deployment.md:499]: A little confusing in the phrasing. Might be better to put it another way, something like: "If IOMMU and VFIO are not enabled in the BIOS, the alternative driver will be used by default." At least that is what I am taking this to mean.

File: docs/admin/deployment.md [hunk 2]
```diff
@@ -748,6 +753,17 @@ For class == "nvme", the following parameters should be populated:
 
 - `bdev_list` should be populated with NVMe PCI addresses.
 
+The default way for DAOS to access NVMe storage is through SPDK via the VFIO user-space driver.
+To use an alternative driver with SPDK, set `disable-vfio: true` in the global section of the config
+file to fallback to using UIO user-space driver with SPDK instead. The option is only effective when
+IOMMU and VFIO are enabled on the host.
```
Comment (kjacque) [inline docs/admin/deployment.md:759]: Same comment as above.

**Review by kjacque — CHANGES_REQUESTED** [review]
One bit of phrasing is confusing to me, otherwise looks good.

File: docs/admin/deployment.md (hunk 1 above)
Comment (tanabarr) [reply docs/admin/deployment.md]: done

File: docs/admin/deployment.md (hunk 2 above)
Comment (tanabarr) [reply docs/admin/deployment.md]: done

**Review by kjacque — APPROVED** [review]

**Review by Michael-Hennecke — APPROVED** [review]

**Pull request merged**